spec/formatting.md

```diff
@@ -320,16 +319,20 @@ Implementation-defined _functions_ SHOULD use an implementation-defined _namespa
 
 #### Option Resolution
 
-The result of resolving _option_ values is an unordered mapping of string identifiers to values.
+The result of resolving _option_ values is an unordered mapping of string identifiers to values,
```

Comment:
Is "unordered" too specific? Most developers know not to make an ordered map unless required to, but we do not need to require it be unordered (and I'm not sure how we'd test that it is unordered).

I'd also suggest tightening the definition:
```suggestion
Resolving _options_ results in a mapping of string _identifiers_ to _values_,
```

Reply:
We agreed specifically about "unordered" in #751, though.

Reply:
You're right: we did discuss that. At that risk of discussing it again, I'd rather say it separately, e.g.

> Resolving _options_ results in a mapping of string _identifiers_ to _values.
> Note that this mapping MAY be unordered.

Reply:
> Note that this mapping MAY be unordered.

I don't think this is testable, not without extrapolation. I think it is cleaner to state this directly in terms of observable differences, something like:

The order among options MUST NOT produce different formatting results. For example,

`.input {$x option1=foo1 option2=foo2}`
must yield exactly the same resulting formats as:
`.input {$x option2=foo2 option1=foo1}`

Reply:
That sounds like a requirement to introduce separately in the Function Handler section.

Reply:
We currently have this requirement (as you pointed out), but it is dimly and unclearly specified here. We already have a consensus for it, so, given where we are in the schedule, I would suggest:

- Add the text @machiatti suggests to Function Handler
- Add the first sentence of my text from the ["you're right" comment](https://github.com/unicode-org/message-format-wg/pull/903#discussion_r1796065115)
- Add a reminding note in place of my "MAY" text

```suggestion
Resolving _options_ results in a mapping of string _identifiers_ to values,
```

And later:

> [!NOTE]
> The order of the _options_ is not significant.

Reply:
I support Addison's suggestion, if done within a follow-on PR. This PR is only adding a comma to the end of this sentence, and the proposed action is introducing a new MUST NOT on function handlers, which are so far only governed by MAY and SHOULD text.

OTOH, if this PR is not merged and accepted on this Monday's call, then sure, let's allow this scope creep and see how far this one PR goes. For example, I note that we say that function handlers SHOULD emit a Bad Operand error, but we don't say the same about Bad Option.